sdk/core/azure-core/test/ut/operation.cpp

```diff
@@ -0,0 +1,56 @@
+// ************ CUSTOMER CODE **********************
+
+#include "gtest/gtest.h"
+
+#include "operation_test.hpp"
+
+#include <azure/core/context.hpp>
+#include <azure/core/operation.hpp>
+
+#include <chrono>
+
+using namespace Azure::Core::Test;
+using namespace std::literals;
+
+TEST(Operation, Poll)
+{
+  StringClient client;
+  auto operation = client.StartStringUpdate();
+
+  EXPECT_FALSE(operation.Done());
+  EXPECT_FALSE(operation.HasValue());
+
+  while(!operation.Done())
+  {
+    EXPECT_FALSE(operation.HasValue());
+    auto response = operation.Poll();
```

Comment:
What does calling `operation.Value()`after this line do? Add an assert for what we expect there.

Reply:
Its an implementation detail for the operation.  Most Operations will throw, but we don't enforce that.

Reply:
Which should probably mandate a specific exception be thrown and put it in the guidelines so that customers know exactly how to recover from this kind of failure. Although, calling Value when HasValue is false is a contract/precondition failure. In other words, a programmer bug, that should not have code written to recover from it. So, perhaps a C++ standard library runtime_error-derived type should be thrown here.

Reply:
> I can add a check for how the test case is implemented. But note that is all that is being verified is how the test was written,

Right, it isn't all that useful testing the test implementation.

> it does not enforce a contract on the Operation

This is what I meant, that we should establish some contract (maybe through documentation) so the implementer of `Operation<T>` knows how to provide an implementation that adheres to the contract so others can call it and get consistent behavior. And then add that test :)